notebooks/203-meter-reader/203-meter-reader.ipynb

```diff
@@ -0,0 +1,730 @@
+    "    roi_stack = np.hstack([resize_imgs[0], resize_imgs[1]])\n",
+    "else:\n",
+    "    roi_stack = resize_imgs[0]\n",
+    "\n",
+    "if cv2.imwrite(f\"{DATA_DIR}/detection_results.jpg\", roi_stack):\n",
+    "    print(\"The detection result image has been saved as \\\"detection_results.jpg\\\" in data\")\n",
+    "    plt.imshow(roi_stack)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Run meter segmentation model\n",
+    "Get the results of segmentation task on detected ROI."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "seg_results = list()\n",
+    "num_imgs = len(roi_imgs)\n",
+    "\n",
+    "# Run meter segmentation model on all meters detected\n",
+    "for i in range(0, num_imgs, seg_batch_size):\n",
+    "    batch = roi_imgs[i : min(num_imgs, i + seg_batch_size)]\n",
+    "    seg_result = predictor([batch], segmenter)\n",
+    "    seg_results.extend(seg_result)\n",
+    "results = []\n",
+    "for i in range(len(seg_results)):\n",
+    "    results.append(np.argmax(seg_results[i], axis=0)) \n",
+    "seg_results = erode(results, erode_kernel)\n",
+    "\n",
+    "# Create the photos of segmentation results\n",
+    "if len(resize_imgs) == 2:\n",
+    "    mask_stack = np.hstack([segmentation_map_to_image(seg_results[0], COLORMAP), \n",
+    "                            segmentation_map_to_image(seg_results[1], COLORMAP)])\n",
+    "else:\n",
+    "    mask_stack = segmentation_map_to_image(seg_results[0], COLORMAP)\n",
+    "\n",
+    "if cv2.imwrite(f\"{DATA_DIR}/segmentation_results.jpg\", mask_stack):\n",
+    "    print(\"The segmentation result image has been saved as \\\"segmentation_results.jpg\\\" in data\")\n",
+    "    plt.imshow(mask_stack)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Postprocess the models result and calculate the final readings\n",
+    "Use OPENCV function to find the location of the poiner in scale map."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Find out the pointer location in scale map and calculate the meters reading \n",
+    "rectangle_meters = circle_to_rectangle(results)\n",
+    "line_scales, line_pointers = rectangle_to_line(rectangle_meters)\n",
+    "binaried_scales = mean_binarization(line_scales)\n",
+    "binaried_pointers = mean_binarization(line_pointers)\n",
+    "scale_locations = locate_scale(binaried_scales)\n",
+    "pointer_locations = locate_pointer(binaried_pointers)\n",
+    "pointed_scales = get_relative_location(scale_locations, pointer_locations)\n",
+    "meter_readings = calculate_reading(pointed_scales)\n",
+    "for i in range(len(meter_readings)):\n",
+    "    print(\"Meter {}: {}\".format(i + 1, meter_readings[i]))\n",
+    "\n",
+    "# Create final result photo with meter value\n",
```

Comment:
It would be nice to show both rectangle_meters here and to move the code below to the next cell.

Reply:
Done